src/extractor/location_dependent_data.cpp

```diff
@@ -0,0 +1,147 @@
+    BOOST_ASSERT(geojson.HasMember("features"));
+    BOOST_ASSERT(geojson["features"].IsArray());
+
+    const auto &features_array = geojson["features"].GetArray();
+    std::vector<rtree_t::value_type> bounding_boxes;
+    for (rapidjson::SizeType i = 0; i < features_array.Size(); i++)
+    {
+        util::validateFeature(features_array[i]);
+        const auto &feature = features_array[i].GetObject();
+        const auto &geometry = feature["geometry"].GetObject();
+        BOOST_ASSERT(geometry.HasMember("type"));
+
+        // Case-sensitive check of type https://tools.ietf.org/html/rfc7946#section-1.4
+        if (std::strcmp(geometry["type"].GetString(), "Polygon"))
+        {
+            util::Log(logDEBUG) << "Skipping non-polygon shape in geojson file";
+            continue;
+        }
+
+        // The first array of polygon coords is the exterior ring
+        // https://tools.ietf.org/html/rfc7946#section-3.1.6
+        polygon_t polygon;
+        const auto &coords_outer_array = geometry["coordinates"].GetArray()[0].GetArray();
+        for (rapidjson::SizeType i = 0; i < coords_outer_array.Size(); ++i)
+        {
+            util::validateCoordinate(coords_outer_array[i]);
+            const auto &coords = coords_outer_array[i].GetArray();
+            polygon.outer().emplace_back(coords[0].GetDouble(), coords[1].GetDouble());
+        }
+        bounding_boxes.emplace_back(boost::geometry::return_envelope<box_t>(polygon),
+                                    polygons.size());
+
+        // Collect feature properties and store in polygons vector
+        auto convert_value = [](const auto &property) -> property_t {
+            if (property.IsString())
+                return std::string(property.GetString());
+            if (property.IsNumber())
+                return property.GetDouble();
+            if (property.IsBool())
+                return property.GetBool();
+            return {};
+        };
+        properties_t properties;
+        for (const auto &property : feature["properties"].GetObject())
+        {
+            properties.insert({property.name.GetString(), convert_value(property.value)});
+        }
+        polygons.push_back(std::make_pair(polygon, properties));
+    }
+
+    // Create R-tree for bounding boxes of collected polygons
+    rtree = rtree_t(bounding_boxes);
+    util::Log() << "Parsed " << polygons.size() << " geojson polygons with location-dependent data";
+}
+
+namespace
+{
+struct table_setter : public boost::static_visitor<>
+{
+    table_setter(sol::table &table, const std::string &key) : table(table), key(key) {}
+    template <typename T> void operator()(const T &value) const { table.set(key, value); }
+    void operator()(const boost::blank &) const { /* ignore */}
+
+    sol::table &table;
+    const std::string &key;
+};
+}
+
+sol::table LocationDependentData::operator()(sol::state &state, const osmium::Way &way) const
```

Comment:
i would like to keep it as a shortcut to avoid creating `unordered_map`. It can be easily adjusted when it will be clear how conflicting tags must be merged